File: 2019/1.cc
#include <fstream>
#include <iostream>
#include <string>

using namespace std;

int calc_fuel(int mass) { return (mass / 3) - 2; }

int calc_all_fuel(std::string filepath) {
  ifstream in(filepath);

  std::string str;
  int total_fuel = 0;
  while (std::getline(in, str)) {
    int mass = std::stoi(str);
    int fuel = calc_fuel(mass);
    cout << "Mass: " << mass << "\tFuel: " << fuel << std::endl;
    total_fuel += fuel;
  }
  return total_fuel;
}

int main() {
  int total_fuel = calc_all_fuel("1_input.txt");
  cout << "Total fuel: " << total_fuel << std::endl;

  return 0;
}

File: 2019/1.py
import math                                                                              

def calc_fuel(mass):
  fuel = math.floor(mass/3) - 2 
  print("Mass: %d\tFuel: %d" % (mass, fuel))
  return fuel

def calc_all_modules(filepath):
  total_fuel = 0 
  with open(filepath, "r") as f:
    for line in f.readlines():
      line = line.strip()
      mass = int(line)
      total_fuel += calc_fuel(mass)


fuel_requirements = calc_all_modules("1_input.txt")
print("Total fuel requirements: %s" % fuel_requirements)
